### PR TITLE
Damir mu

### PR DESCRIPTION
**Pull request by Damir-Mu**

### CODE REVIEW

File: src/components/Chats/Chats.jsx [hunk 1]
```diff
@@ -111,19 +112,24 @@ export const Chats = () => {
...
               <span>{chat[1]?.userInfo?.displayName}</span>
               {/* <span>{chat[1]?.userInfo?.isOnline ? "Online" : "Offline"}</span> */}
               <p>{chat[1]?.lastMessage?.text}</p>
+              {chat[1]?.userInfo?.online && (
+                <span className="onlineStatus">Online</span>
+              )}
+              {!chat[1]?.userInfo?.online && (
+                <span className="onlineStatus">offline</span>
+              )}
             </div>
```
Comment (DenisTagaev) [inline src/components/Chats/Chats.jsx]: Check mobile view for the spans

File: src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -23,9 +23,9 @@ export const Navbar = () => {
             <p>Profile</p>
           </div>
         </div>
-        <button onClick={async() => {
+        <button onClick={async () => {
           await updateDoc(doc(db, "users", currentUser.uid), {
-              online: false,
+            online: false,
           });
```
Comment (DenisTagaev) [inline src/components/Navbar/Navbar.jsx:29]: ???

File: src/components/context/AuthContext.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ export const AuthContextProvider = ({ children }) => {
         //trigger updates across all components
         const unsubscribe = onAuthStateChanged(auth, async (user) => {
             setCurrentUser(user);
-            if(user) {
+            if (user) {
                 await updateDoc(doc(db, "users", user.uid), {
```
Comment (DenisTagaev) [inline src/components/context/AuthContext.js:21]: A lot of extra spaces, not really necessary, maybe you should check your formatter settings.

File: src/pages/profile/AuthTab/AuthTab.jsx [hunk 4]
```diff
@@ -50,22 +47,17 @@ export default function AuthTab() {
...
+                formData.password
+            );
+            // authenticating the user 
+            await reauthenticateWithCredential(currentUser, credential).then((res) => {
+                // User re-authenticated.
+                console.log(res)
+                // opening the input fields
+                setPasswordChangeAccess(!passwordChangeAccess);
+            }).catch((error) => { alert(error.message) });
         }
```
Comment (DenisTagaev) [inline src/pages/profile/AuthTab/AuthTab.jsx:61]: Good job, much cleaner code!

File: src/pages/profile/AuthTab/AuthTab.jsx [hunk 5]
```diff
@@ -128,31 +120,22 @@ export default function AuthTab() {
...
+                            type="password"
+                            name="password"
+                            value={formData.password}
+                            placeholder="Enter current password"
+                            onChange={handlePasswordChange}
+                        />
+                        {formErrors.password && (
+                            <span className="formError">{formErrors.password}</span>
+                        )}
+                        <button className="iconBtn" type="submit" disabled={formErrors.email || formErrors.password}>Log in<AiFillEdit /></button>
```
Comment (DenisTagaev) [inline src/pages/profile/AuthTab/AuthTab.jsx]: You sure about `formErrors.email`? Where are you setting this error?

File: src/pages/profile/ProfileTab/ProfileTab.jsx [hunk 6]
```diff
@@ -177,8 +146,9 @@ export default function ProfileTab() {
         event.preventDefault();
 
         if (!nameError) {
-            await updateProfile(currentUser, { displayName: userName });
-            console.log(currentUser)
+            await updateDoc(doc(db, "users", currentUser.uid), { displayName: userName });
+            await updateProfile(currentUser, { displayName: userName })
+            console.log(currentUser.displayName + "document updated")
             setChangeNameAccess(!changeNameAccess);
         }
```
Comment (DenisTagaev) [inline src/pages/profile/ProfileTab/ProfileTab.jsx:157]: Commented

**Review by DenisTagaev — CHANGES_REQUESTED** [review]

File: src/pages/profile/AuthTab/AuthTab.jsx (hunk 4 above)
Comment (Damir-Mu) [reply src/pages/profile/AuthTab/AuthTab.jsx:61]: Thank you, you are a mentor figure

File: src/pages/profile/ProfileTab/ProfileTab.jsx [hunk 7]
```diff
@@ -51,7 +51,7 @@ export default function ProfileTab() {
             const unsub = getDoc(doc(db, "users", currentUser.uid)).then((doc) => {
                 const data = doc.data();
                 setUserName(data.displayName)
-                setDesc(data.profileDescription);
+                setDesc(data.profileDescription ? data.profileDescription : formDesc);
             });
```
Comment (DenisTagaev) [inline src/pages/profile/ProfileTab/ProfileTab.jsx:55]: data.profileDescription ? setDesc(data.profileDescription) : setDesc(formDesc);

File: src/pages/profile/ProfileTab/ProfileTab.jsx [hunk 8]
```diff
@@ -31,7 +31,7 @@ export default function ProfileTab() {
         maritalStatus: "",
     })
 
-    const [desc, setDesc] = useState({});
+    const [desc, setDesc] = useState(formDesc);
     const [descriptionChangeAccess, setDescriptionChangeAccess] = useState(false);
```
Comment (DenisTagaev) [inline src/pages/profile/ProfileTab/ProfileTab.jsx:35]: useState({})

**Review by DenisTagaev — CHANGES_REQUESTED** [review]

**Review by DenisTagaev — APPROVED** [review]

**Pull request merged**